common/common-api/src/main/java/org/ow2/proactive/utils/Lambda.java

```diff
@@ -64,6 +64,59 @@ public static void withLock(Lock lock, Runnable runnable) {
         }
     }
 
+    public static <T, E extends Exception> T withLockInterruptible(Lock lock, Callable<T> callable,
+            E interruptedException) throws E {
+
+        try {
+            lock.lockInterruptibly();
+            return callable.call();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            if (interruptedException != null) {
+                interruptedException.initCause(e);
+                throw interruptedException;
+            }
+            throw new RuntimeException(e);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    public static <T, E1 extends Exception> T withLockException1(Lock lock, Callable<T> callable, Class<E1> type1)
```

Comment:
Function could be renamed as `withLockNotWrappingException(lock, callable, notWrappingExceptionType)`